dataland-internal-storage/src/main/kotlin/org/dataland/datalandinternalstorage/services/DatabaseStringDataStore.kt

```diff
@@ -103,24 +109,79 @@ class DatabaseStringDataStore(
         payload: String,
     ) {
         logger.info("Received DataID $dataId and CorrelationId: $correlationId")
-        val data = temporarilyCachedDataClient.getReceivedPublicData(dataId)
+        val dataObject = temporarilyCachedDataClient.getReceivedPublicData(dataId)
         logger.info("Inserting data into database with data ID: $dataId and correlation ID: $correlationId.")
-        storeDataItemWithoutTransaction(DataItem(dataId, objectMapper.writeValueAsString(data)))
+
+        val dataAsJSON = JSONObject(dataObject)
```

Comment:
This entirely breaks our current assumption of the Internal Storage: So far, internal storage always was a "dumb" Key-Value Store. IMO we should carefully consider implement such logic here.